src/command.cc

```diff
@@ -198,6 +199,56 @@ AttrPathMixin::fixupAttrPath()
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+  argparse::Argument &
+RegistryFileMixin::addRegistryFileArg( argparse::ArgumentParser & parser )
+{
+  return parser.add_argument( "--registry-file" )
+               .help(
+                  "The path to the 'registry.json' file."
+               )
+               .required()
+               .metavar( "PATH" )
+               .action( [&]( const std::string & strPath )
+                        {
+                          this->setRegistryPath( strPath );
+                        }
+                      );
+}
+
+  void
+RegistryFileMixin::setRegistryPath(const std::filesystem::path & path)
+{
+  this->registryPath = path;
+}
+
+  const RegistryRaw &
+RegistryFileMixin::getRegistryRaw()
+{
+  if ( this->registryRaw.has_value() )
+    {
+      return * this->registryRaw;
+    }
+  this->loadRegistry();
+  return * this->registryRaw;
+}
+
+  void
+RegistryFileMixin::loadRegistry()
+{
+  if ( !this->registryPath.has_value() )
```

Comment:
suggestion: either handle empty string here or in `setRegistryPath` (and keep the setter 😛 )

It also seems an error like this should be surfaced in `action()` and not at any other time

```suggestion
  if ( !this->registryPath.has_value() && !this->registryPath->empty())
```

Reply:
Fixed